editor/src/clj/editor/app_view.clj

```diff
@@ -828,21 +838,23 @@
        (targets/controllable-target? target)
        (targets/remote-target? target)))
 
-(defn- launch-built-project! [project engine-descriptor project-directory prefs web-server debug?]
+(defn- launch-built-project! [project engine-descriptor project-directory prefs web-server debug? workspace]
   (let [selected-target (targets/selected-target prefs)
         launch-new-engine! (fn []
                              (targets/kill-launched-targets!)
-                             (let [launched-target (launch-engine! engine-descriptor project-directory prefs debug?)
-                                   log-stream      (:log-stream launched-target)]
-                               (targets/when-url (:id launched-target)
-                                                 #(on-launched-hook! project (:process launched-target) %))
-                               (reset-console-stream! log-stream)
-                               (reset-remote-log-pump-thread! nil)
-                               (start-log-pump! log-stream (make-launched-log-sink launched-target))
-                               launched-target))]
+                             (let [launched-targets (launch-engine! engine-descriptor project-directory prefs debug? workspace)
+                                   last-launched-target (last launched-targets)]
+                               (doseq [launched-target launched-targets]
+                                 (targets/when-url (:id launched-target)
+                                                   #(on-launched-hook! project (:process launched-target) %))
+                                 (let [log-stream (:log-stream launched-target)]
+                                   (reset-console-stream! log-stream)
+                                   (reset-remote-log-pump-thread! nil)
+                                   (start-log-pump! log-stream (make-launched-log-sink launched-target))))
```

Comment:
we get address and the other useful info from the first few lines in logs. So, we have to do it for each target we launch, but keep console connected only to the last target